update-docker-tags.go

```diff
@@ -45,6 +47,8 @@ Examples:
 		os.Exit(2)
 	}
 	flag.Var(&constraintArgs, "constraint", "(repeatable) add a semver constraint for a given docker image")
+	useGeneric := flag.Bool("generic", false, "generic matches additional docker image tags")
+	customPattern := flag.String("tag-pattern", "", "specify a custom regexp to match docker image tags")
```

Comment:
```suggestion
	customPattern := flag.String("pattern", "", "specify a custom regexp to match docker images")
```

It's not just matching tags, right? It's matching the whole image repository itself

Reply:
Instead of having both `-generic` and `-tag-pattern`, can we just have a single `-pattern` which defaults to the "generic" pattern you have in `GENERIC_TAG_PATTERN` right now?

I think that would be most ideal

Reply:
That makes more sense

Reply:
I did run into a bit of an issue here. The flag is defined as a string and will take in the path as the regex if it is defined as just `update-docker-tags -pattern dir/`. It will try to interpret the path as part of flag. There isn't really a clean workaround here without importing a library or hacking around the flag

Reply:
@daxmc99 I think this is still an issue.

I'm not sure I understand your example of the problem with doing this, though? Why would you specify `-pattern` without an argument? If you want the generic pattern, you just wouldn't specify `-pattern` at all

Reply:
This code has been changed so that there is no sourcegraph specific pattern and now `-pattern` is only used now to pass a custom regex